crates/sui-core/src/authority_active/gossip/mod.rs

```diff
@@ -208,11 +231,77 @@ where
     })
 }
 
-impl<A> PeerGossip<A>
+#[async_trait]
+trait FollowerDigestHandler<A> {
+    async fn handle_digest(&self, follower: &Follower<A>, digest: ExecutionDigests) -> SuiResult;
```

Comment:
We can also make the input to this function a queue, rather than the individual digests, to make it easier to keep track of when the quorum of certificates has been seen, but that can be extended later.

Reply:
One of the future todo items is to put all the stream contents from all the peer validator batch streams into a single queue to ensure we don't download duplicates at the same time, it sounds like this will also assist in keeping track of quorums.

Reply:
I'm doing exactly that in another branch right now, I will see if I can easily make the de-duplication step usable for gossip as well.